src/ui/components/shared/LoadingScreen.tsx

```diff
@@ -28,49 +29,67 @@ export function LoadingScreenTemplate({ children }: { children?: ReactNode }) {
   }, [hoverboardColor]);
 
   return (
-    <BubbleViewportWrapper>
-      <div className="relative flex w-96 flex-col items-center space-y-1 rounded-lg bg-loadingBoxes p-4 py-12 shadow-sm">
-        <div className="flex flex-col items-center space-y-2">
-          <div className="h-32 w-32" onClick={changeHoverboardColor}>
-            <Hoverboard color={hoverboardColor} />
+    <div className={styles.loadingScreenWrapper}>
+      <BubbleViewportWrapper>
+        <div className={styles.loadingScreenWrapper}>
+          <div className="flex flex-col items-center space-y-2">
+            <div className={styles.hoverboardWrapper} onClick={changeHoverboardColor}>
+              <Hoverboard color={hoverboardColor} />
+            </div>
+            {children}
           </div>
-          {children}
         </div>
-      </div>
-    </BubbleViewportWrapper>
+      </BubbleViewportWrapper>
+    </div>
   );
 }
 
+const phrases = [
+  "Tuning the flux capacitor...",
+  "Prepping DeLorean for time travel...",
+  "Revving up to 88 miles per hour...",
+  "Gathering 1.21 gigawatts of power...",
+  "Ensuring temporal paradox safeguards...",
+  "Being careful not to mess up your timeline...",
+];
+
 function LoadingScreen({
   uploading,
   awaitingSourcemaps,
   fallbackMessage,
-  stalledTimeout = 9000,
+  stalledTimeout = 15000,
 }: PropsFromRedux & { fallbackMessage: string; stalledTimeout?: number }) {
-  // The backend send events in this order: uploading replay -> uploading sourcemaps.
-  let waitingForMessage = <span>{fallbackMessage}</span>;
-  if (awaitingSourcemaps) {
-    waitingForMessage = <span>Uploading sourcemaps...</span>;
-    stalledTimeout = Infinity;
-  } else if (uploading) {
-    waitingForMessage = <span>Uploading {Math.round(+uploading.amount)}Mb</span>;
-    stalledTimeout = Infinity;
-  }
-
-  let [stalled, setStalled] = useState<boolean>(false);
-  useEffect(() => {
-    if (stalledTimeout === Infinity) {
-      return;
-    }
+  const longWaitMessage =
+    "<div><p>This is taking longer than expected.</p><p><a href='http://replay.io/discord' target='discord'>Contact us on Discord</a></p>";
+
+  const [message, setMessage] = useState(fallbackMessage);
 
-    const timeout = setTimeout(() => setStalled(true), stalledTimeout);
+  const changeMessage = useCallback(() => {
+    const randomIndex = Math.floor(Math.random() * phrases.length);
+    setMessage(phrases[randomIndex]);
+  }, []);
 
-    return () => clearTimeout(timeout);
-  }, [fallbackMessage, stalledTimeout]);
+  useEffect(() => {
+    const phraseTimeout = setTimeout(changeMessage, 5000);
+    return () => clearTimeout(phraseTimeout);
+  }, [changeMessage]);
```

Comment:
Since the effect is the only code that uses `changeMessage`, you could simplify this:
```js
useEffect(() => {
  const changeMessage = () => {
    // Nit that it's possible to get the same index twice this way;
    // A more robust way of doing this would be sorting the array once (during module initialization?)
    // and then just stepping through in-order (`index = (index + 1) % length`)
    const randomIndex = Math.floor(Math.random() * phrases.length);
    setMessage(phrases[randomIndex]);
  };

  const phraseTimeout = setTimeout(changeMessage, 5000);

  return () => clearTimeout(phraseTimeout);
}, []);
```